src/Magento/FunctionalTestingFramework/Test/Util/TestObjectExtractor.php

```diff
@@ -118,6 +118,21 @@ public function extractTestData($testData)
         //Override features with module name if present, populates it otherwise
         $testAnnotations["features"] = [$module];
 
+        // Append test file names in description annotation
+        if (!empty($fileNames)) {
+            if (isset($testAnnotations["description"][0])) {
```

Comment:
Need to check if `description` array is set at all and set it. You can also simplify these conditionals:

```
if (!empty($fileNames)) {
    if (!isset($testAnnotations["description"])) {
        $testAnnotations["description"] = [];
    }
    $description = $testAnnotations["description"][0] ?? '';
    $testAnnotations["description"][0] = $this->appendFileNamesInDescriptionAnnotation($description, $fileNames);
}
```

Reply:
My code just set the array value directly regardless of `description` array is set or not, but your code looks more strait forward. I will double check and change it if it works.